File: ABC/352/a.cpp
#include <iostream>
using namespace std;

int main() {
    int n, x, y, z;
    cin >> n >> x >> y >> z;

    if (x > y) {
        if (y < z && z < x) cout << "Yes" << endl;
        else cout << "No" << endl;
    } else {
        if (x < z && z < y) cout << "Yes" << endl;
        else cout << "No" << endl;
    }

    return 0;
}

File: ABC/352/a.cs
using System;

class Program
{
    static void Main(string[] args)
    {
        string[] input = Console.ReadLine().Split(' ');
        int n = int.Parse(input[0]);
        int x = int.Parse(input[1]);
        int y = int.Parse(input[2]);
        int z = int.Parse(input[3]);

        if ((y < z && z < x) || (x < z && z < y))
        {
            Console.WriteLine("Yes");
        }
        else
        {
            Console.WriteLine("No");
        }
    }
}
